src/System.Globalization.Extensions/tests/IdnMapping/IdnMappingGetAsciiTests.cs

```diff
@@ -83,24 +83,29 @@ public void GetAscii(string unicode, int index, int count, string expected)
             }
             Assert.Equal(expected, new IdnMapping().GetAscii(unicode, index, count));
         }
-        
-        [Fact]
-        public void TestGetAsciiWithDot()
+
+        public static bool GetAsciiWithDotSucceeds => !GetAsciiWithDotFails;
+
+        [ConditionalFact(nameof(GetAsciiWithDotSucceeds))]
+        public void TestGetAsciiWithDot_Succeeds()
         {
-            string result = "";
-            Exception ex = Record.Exception(()=> result = new IdnMapping().GetAscii("."));
-            
-            if (ex == null)
-            {
-                // Windows and OSX always throw exception. some versions of Linux succeed and others throw exception   
-                Assert.False(RuntimeInformation.IsOSPlatform(OSPlatform.Windows));
-                Assert.False(RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
-                Assert.Equal(result, ".");
-            }
-            else
-            {
-                Assert.True(ex is ArgumentException);
-            }
+            string dot = ".";
+            string result = new IdnMapping().GetAscii(dot);
+            Assert.Equal(dot, result);
+        }
+
+        public static bool GetAsciiWithDotFails =>
+            PlatformDetection.IsWindows ||
+            PlatformDetection.IsOSX ||
+            PlatformDetection.IsUbuntu1604 ||
+            PlatformDetection.IsUbuntu1610 ||
+            PlatformDetection.IsOpenSUSE;
```

Comment:
@stephentoub - Maybe.
The catch with Ubuntu is, 16 fails the call, but 14 succeeds, which is already potentially an issue.  Probably then I should flip it to say "Ubuntu, but not 14".
Also, given #8346 , the behavior of OpenSUSE has changed (maybe important), which we would never have noticed with the original test.